src/OpenSpaceToolkit/Astrodynamics/Trajectory/Segment.cpp

```diff
@@ -101,6 +105,108 @@ Mass Segment::Solution::computeDeltaMass() const
     return Mass::Kilograms(getInitialMass().inKilograms() - getFinalMass().inKilograms());
 }
 
+MatrixXd Segment::Solution::getDynamicsContribution(
+    const Shared<Dynamics>& aDynamicsSPtr,
+    const Shared<const Frame>& aFrameSPtr,
+    const Array<Shared<const CoordinatesSubset>>& aCoordinatesSubsetSPtrArray
+) const
+{
+    // Check dynamics is part of the segment dynamics
+    if (!dynamics.contains(aDynamicsSPtr))
+    {
+        throw ostk::core::error::RuntimeError("Provided dynamics is not part of the segment dynamics.");
+    }
+
+    // Extract write coordinates subsets from dynamics
+    const Array<Shared<const CoordinatesSubset>> dynamicsWriteCoordinatesSubsets =
+        aDynamicsSPtr->getWriteCoordinatesSubsets();
+
+    // Check that the provided coordinates subsets are part of the dynamics write coordinates subsets
+    for (auto aCoordinatesSubsetSPtr : aCoordinatesSubsetSPtrArray)
+    {
+        if (!dynamicsWriteCoordinatesSubsets.contains(aCoordinatesSubsetSPtr))
+        {
+            throw ostk::core::error::RuntimeError(
+                "Provided coordinates subset is not part of the dynamics write coordinates subsets."
+            );
+        }
+    }
+
+    // Initialize the definitive coordinate subset array
+    Array<Shared<const CoordinatesSubset>> definitiveCoordinateSubsetArray = aCoordinatesSubsetSPtrArray;
+
+    // Check value for aCoordinatesSubsetSPtrArray
+    if (aCoordinatesSubsetSPtrArray.isEmpty())
+    {
+        definitiveCoordinateSubsetArray = aDynamicsSPtr->getWriteCoordinatesSubsets();
+    }
+
+    // Extract states size
+    const Size stateSize = this->states.getSize();
```

Comment:
This is the number of rows right? (i.e. number of states?)

Just asking as `stateSize` sound like the size of the state (i.e. number of columns)

Reply:
Good point, that's confusing adapting